Scripts/Helpers/PlayerFloor.cs

```diff
@@ -0,0 +1,35 @@
+﻿#if UNITY_EDITOR
+using UnityEngine;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEditor.Experimental.EditorVR.Core;
+using UnityEditor.Experimental.EditorVR.Extensions;
+
+namespace UnityEditor.Experimental.EditorVR.Helpers
+{
+    [ExecuteInEditMode]
+    public class PlayerFloor : MonoBehaviour, IUsesViewerScale
+    {
+        Vector3 floorPosition;
+        Quaternion floorRotation;
+        Transform m_Camera;
+        Transform m_CameraRig;
+
+        private void Awake()
+        {
+            m_Camera = CameraUtils.GetMainCamera().transform;
+            m_CameraRig = CameraUtils.GetCameraRig();
+        }
+
+        void Update()
+        {
+            floorPosition.x = m_Camera.position.x + 0.05f * this.GetViewerScale();
```

Comment:
cache viewer scale value initially, before utilizing it 3X below

Reply:
I got an error saying that the contextual keyword 'var' may only appear within a local variable when I wrote "var currentScale = this.GetViewerScale()". Should I just declare GetViewScale() instead?

Reply:
Maybe you were using "var" inline with the assignment? It should work if you do
```var currentScale = this.GetViewerScale();
floorposition.x = ... * currentScale```